### DISCUSSION TITLE
Creating or picking transition?

### DISCUSSION THREAD

**Discussion by omar2205** [material-components/material-web#5579] [Q&A] [answered]
I have this simple `md-dialog`, how to pick what transition it uses?

```html
<script type="module">
  import 'https://esm.run/@material/web@1.4.0/all.js'
</script>

<md-outlined-button class="btn">Back</md-outlined-button>

<md-dialog class="da" fade>
  <div slot="headline">
    Dialog title
  </div>
  <form slot="content" id="form-id" method="dialog">
    A simple dialog with free-form content.
  </form>
  <div slot="actions">
    <md-text-button form="form-id">Ok</md-text-button>
  </div>
</md-dialog>
```

It doesn't say much here
![image](https://github.com/material-components/material-web/assets/1373867/28860c82-9e02-4a11-a423-638ed2eaba7d)

**Comment by asyncliz** [comment, marked as answer]
You can customize the transition by overriding those functions and returning your own animation keyframes to change any aspect of the animations.

There's a [default open/close animation object](https://github.com/material-components/material-web/blob/69b4e9472b7c8234b9d1b2462fcb843d8d409ee5/dialog/internal/animations.ts#L52) that you can reference for the options available.

Each part of the dialog (like the dialog itself, the scrim, or the content) can be given one or more animations, represented as an array of [`Element.animate()` arguments](https://developer.mozilla.org/en-US/docs/Web/API/Element/animate).

```js
const dialog = document.querySelector('md-dialog');
dialog.getOpenAnimation = () => {
  // A simpler and faster fade-in animation
  return {
    // An array of one or more animations
    dialog: [
      // A single simple fade-in animation
      [
        // args for Element.animate(keyframes, options);
        [{'opacity': '0'}, {'opacity': '1'}],
        {duration: 250, easing: 'ease-in-out'},
      ]
    ],
    scrim: [
      // Same fade-in for the scrim
      [
        [{'opacity': '0'}, {'opacity': '0.32'}],
        {duration: 250, easing: 'ease-in-out'},
      ]
    ],
  };
};
```

**Reply by omar2205** [reply]
I tried and it didn't work https://codepen.io/omar2205/pen/yLrEmye

\+ Was hoping a section about on https://material-web.dev/

**Reply by asyncliz** [reply]
It looks like the property is being set too early before the element is registered. Try waiting for the element to register:

```js
customElements.whenDefined('md-dialog').then(() => {
  const about_dialog = document.querySelector('.about');
  about_dialog.getOpenAnimation = ...;
});
```

This is needed when the scripts that modify the element run before the script that imports the element.

**Reply by omar2205** [reply]
That worked, thanks!
